File: newcoder/question/NC4.cpp
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode* fast = head;
        ListNode* low = head;
        if(head == NULL){
            return false;
        }
        if(head -> next == NULL){
            return false;
        }
        
        while(true){
            if(fast == NULL || fast -> next == NULL){
                return false;
            }
            fast = fast -> next -> next;
            low = low -> next;
            if(fast == low){
                return true;
            }
        }
        return false;
    }
};

File: nowcoder/question/NC4.py
class Solution:
    def hasCycle(self , head ):
        # write code here
        fast = head;
        low = head;
        if head == None or head.next == None:
            return False
        while True:
            if(fast != None and fast.next != None):
                fast = fast.next.next
            else:
                return False
            low = low.next
            if(fast == low):
                return True